server/golden/proxy.go

```diff
@@ -0,0 +1,467 @@
+	if err != nil {
+		return err
+	}
+	conn.Entry = conn.Entry.WithField("query", query)
+
+	_, err = h.processQuery(c, conn, query, false, callback)
+	if err != nil {
+		conn.Errorf("Failed to process MySQL results: %s", err)
+	}
+	return err
+}
+
+func (h MySqlProxy) processQuery(
+	c *mysql.Conn,
+	proxy proxyConn,
+	query string,
+	isMultiStatement bool,
+	callback func(*sqltypes.Result, bool) error,
+) (string, error) {
+	ctx := sql.NewContext(h.ctx)
+	var remainder string
+	if isMultiStatement {
+		_, query, remainder, _ = parse.ParseOne(ctx, query)
+	}
+
+	ctx = ctx.WithQuery(query)
+	more := remainder != ""
+
+	proxy.Debugf("Sending query to MySQL")
+	rows, err := proxy.Query(query)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		if cerr := rows.Close(); cerr != nil {
+			err = cerr
+		}
+	}()
+
+	var proccesedAtLeastOneBatch bool
+	res := &sqltypes.Result{}
+	ok := true
+	for ok {
+		if res, ok, err = fetchMySqlRows(ctx, rows, 128); err != nil {
+			return "", err
+		}
+		if err := callback(res, more); err != nil {
+			return "", err
+		}
+		proccesedAtLeastOneBatch = true
+	}
+
+	if err := setConnStatusFlags(ctx, c); err != nil {
+		return remainder, err
+	}
+
+	switch len(res.Rows) {
+	case 0:
+		if len(res.Info) > 0 {
+			ctx.GetLogger().Tracef("returning result %s", res.Info)
+		} else {
+			ctx.GetLogger().Tracef("returning empty result")
+		}
+	case 1:
+		ctx.GetLogger().Tracef("returning result %v", res)
+	}
+
+	// processedAtLeastOneBatch means we already called resultsCB() at least
+	// once, so no need to call it if RowsAffected == 0.
+	if res != nil && (res.RowsAffected == 0 && proccesedAtLeastOneBatch) {
+		return remainder, nil
+	}
+
+	return remainder, nil
+}
+
+// WarningCount is called at the end of each query to obtain
+// the value to be returned to the client in the EOF packet.
+// Note that this will be called either in the context of the
+// ComQuery resultsCB if the result does not contain any fields,
+// or after the last ComQuery call completes.
+func (h MySqlProxy) WarningCount(c *mysql.Conn) uint16 {
+	return 0
+}
+
+// See https://dev.mysql.com/doc/internals/en/status-flags.html
+func setConnStatusFlags(ctx *sql.Context, c *mysql.Conn) error {
+	ok, err := isSessionAutocommit(ctx)
+	if err != nil {
+		return err
+	}
+	if ok {
+		c.StatusFlags |= uint16(mysql.ServerStatusAutocommit)
+	} else {
+		c.StatusFlags &= ^uint16(mysql.ServerStatusAutocommit)
+	}
+	if t := ctx.GetTransaction(); t != nil {
+		c.StatusFlags |= uint16(mysql.ServerInTransaction)
+	} else {
+		c.StatusFlags &= ^uint16(mysql.ServerInTransaction)
+	}
+	return nil
+}
+
+func isSessionAutocommit(ctx *sql.Context) (bool, error) {
+	autoCommitSessionVar, err := ctx.GetSessionVariable(ctx, sql.AutoCommitSessionVar)
+	if err != nil {
+		return false, err
+	}
+	return sql.ConvertToBool(autoCommitSessionVar)
+}
+
+func fetchMySqlRows(ctx *sql.Context, results *dsql.Rows, count int) (res *sqltypes.Result, more bool, err error) {
```

Comment:
Totally optional, but you could have `count` be some hidden system variable, so that it's tweakable at runtime.